packages/client/src/components/pages/muster-page/muster-page.tsx

```diff
@@ -224,25 +217,53 @@ export const MusterPage = () => {
     }
   }, [rosterFromDateIso, rosterToDateIso, rosterRowsPerPage, rosterPage, orgId, rosterUnitId, showErrorDialog]);
 
-  const reloadTrendData = useCallback(async () => {
+  const reloadMusterComplianceData = useCallback(async () => {
     try {
-      const data = await MusterClient.getMusterUnitTrends(orgId, {
-        currentDate: moment().toISOString(),
-        weeksCount: '6',
-        monthsCount: '6',
-      });
-      setRawTrendData(data);
+      // determine if we are displaying for all units or just a single unit
+      const unitsToDisplay = rosterUnitId === -1 ? units : [units.find(unit => unit.id === rosterUnitId)];
+      const plotData = new Map();
+      const overallComplianceData = new Map();
+      const todaysComplianceData = new Map();
+      for (let i = 0; i < unitsToDisplay.length; i++) {
```

Comment:
Can be replaced with **forEach** and not to worry about the index "i".

Reply:
This is probably part of my own paranoia but in general when looping and calling async functions inside the loop I do NOT use higher order functions like this. see: https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop